doc/asic_thermal_monitoring_hld.md

```diff
@@ -0,0 +1,255 @@
+
+        Returns:
+            A list of objects derived from ThermalBase representing all thermals
+            available on this module
+        """
+        return self._thermal_list
+
+    def get_thermal(self, index):
+        """
+        Retrieves thermal unit represented by (0-based) index <index>
+
+        Args:
+            index: An integer, the index (0-based) of the thermal to
+            retrieve
+
+        Returns:
+            An object dervied from ThermalBase representing the specified thermal
+        """
+        thermal = None
+
+        try:
+            thermal = self._thermal_list[index]
+        except IndexError:
+            sys.stderr.write("THERMAL index {} out of range (0-{})\n".format(
+                             index, len(self._thermal_list)-1))
+
+        return thermal
+
+    def get_average_temperature(self):
+        """
+        Retrieves the average temperature reading from thermal
+
+        Returns:
+            A float number of average temperature in Celsius up to nearest thousandth
+            of one degree Celsius, e.g. 30.125
+        """
+        raise NotImplementedError
+
+    def get_maximum_temperature(self):
+        """
+        Retrieves the maximum temperature reading from thermal
+
+        Returns:
+            A float number of average temperature in Celsius up to nearest thousandth
+            of one degree Celsius, e.g. 30.125
+        """
+        raise NotImplementedError
+
+```
+
+The chassis_base and module_base of the sonic_platform package now contain the following additional lists and methods:
+
+```
+    # List of AsicBase-derived objects representing all ASICs
+    # available on the chassis/module
+    _asic_list = []
+
+    ##############################################
+    # Asic methods
+    ##############################################
+
+    def get_num_asics(self):
+        """
+        Retrieves the number of ASICs available on this chassis/module
+
+        Returns:
+            An integer, the number of ASIC modules available on this chassis/module
+        """
+        return len(self._asic_list)
+
+    def get_all_asics(self):
+        """
+        Retrieves all ASIC modules available on this chassis/module
+
+        Returns:
+            A list of objects derived from AsicBase representing all ASIC
+            modules available on this chassis/module
+        """
+        return self._asic_list
+
+    def get_asic(self, index):
+        """
+        Retrieves ASIC module represented by (0-based) index <index>
+
+        Args:
+            index: An integer, the index (0-based) of the ASIC module to
+            retrieve
+
+        Returns:
+            An object dervied from AsicBase representing the specified ASIC
+            module
+        """
+
+```
+
+The ThermalBase() implementation for ASIC sensors will return the latest values exported by the SwitchOrch to the stateDB's TEMPERATURE_INFO table.

```

Comment:
looks strange to me, the current idea is that platform API responsible to get the thermal sensor info from the lower layer and populate it to DB, and state DB info will feed CLI/SNMP or telemetry etc. but here you are suggesting platform API to read sensor info from DB?

Reply:
The objective is to present ASIC internal sensors to any advanced thermal control algorithm. The reason it appears dissimilar to the standard platform sensors is because the intent was not to let pmon (or any platform daemon) make SAI calls (and create a new dependency) since this is a unique case. Hence letting the poller be part of SwitchOrch.